bindings/nodejs/lib/types/block/payload/transaction/essence.ts

```diff
@@ -16,56 +19,67 @@ enum TransactionEssenceType {
 }
 
 abstract class TransactionEssence {
-    private type: TransactionEssenceType;
+    readonly type: TransactionEssenceType;
 
     constructor(type: TransactionEssenceType) {
         this.type = type;
     }
-
-    /**
-     * The type of essence.
-     */
-    getType(): TransactionEssenceType {
-        return this.type;
-    }
 }
 
 /**
  * RegularTransactionEssence transaction essence.
  */
 class RegularTransactionEssence extends TransactionEssence {
-    /// The unique value denoting whether the block was meant for mainnet, testnet, or a private network.
-    networkId: number;
-    inputsCommitment: HexEncodedString;
+    /**
+     * The unique value denoting whether the block was meant for mainnet, testnet, or a private network.
+     */
+    networkId: u64;
+
+    @Type(() => Input, {
+        discriminator: ContextInputDiscriminator,
+    })
+    contextInput: ContextInput[];
```

Comment:
> Context Inputs **optAnyOf**

In Rust it has the type `Vec<ContextInput>` so it's not under `Option`. In case it's not provided it will be an empty array? I kept it the same here, but we could possibly make it an optional property `contextInput?: contextInput[]` Not sure how serialization would work though.

Reply:
Keep in mind that after compilation none of these types exist anymore, itll be `undefined` probably if left completely out of the json. Thats fine though